hs_core/views/__init__.py

```diff
@@ -670,12 +672,22 @@ def _unshare_resource_with_users(request, requesting_user, users_to_unshare_with
     return go_to_resource_listing_page
 
 
-def _set_resource_sharing_status(request, user, resource, is_public):
+def _set_resource_sharing_status(request, user, resource, is_public, is_discoverable=False):
+    if not user.uaccess.can_change_resource_flags(resource):
+        messages.error(request, "You don't have permission to change resource sharing status")
+        return
+
     if is_public and not resource.can_be_public:
         messages.error(request, "Resource may not have sufficient required metadata to be public")
     else:
-        if user.uaccess.can_change_resource_flags(resource):
-            resource.raccess.public = is_public
-            resource.raccess.save()
+        if is_discoverable:
```

Comment:
@pkdash for migration of existing resources from old system to new system, after resource.raccess.public is set to True, is discoverable set to True as well automatically by access control? In other words, do I need to set discoverable property properly when migrating resources over? Currently, I only check whether the old resources are public or not, if they are public, res.raccess.public is set to True. Is this sufficient for old resources to work properly in the new system?